### ISSUE TITLE
Invalid rule violation when import is used only for type hints (flake8-bandit S404)

### ISSUE THREAD

**Issue by divaltor** [astral-sh/ruff#9702]
Hello, thanks for Ruff again! I noticed that enabling that rule can cause (probably) false-positive violations which I can't say with guarantee they're false-positive, but probably they are

Of course I can just disable that rule in the configuration file, but may be it should be discussed here about ignoring such cases

```py
from __future__ import annotations

from subprocess import Popen

import ffmpeg

# ffmpeg here is library https://github.com/kkroening/ffmpeg-python
# Popen here is just type hint
process: Popen = ffmpeg.run_async(...)
```

Ruff 0.1.15

```bash
ruff --isolated --preview --select S404 --target-version py312

> S404 `subprocess` module is possibly insecure
```

Also it violates on such code

```py
from subprocess import TimeoutExpired

import ffmpeg

try:
     ffmpeg.run(...)
except TimeoutExpired:
     ...
```

**Comment by divaltor** [comment]
> Hm this rule just seems way overzealous but there are a bunch of rules in flake8-bandit that just flag imports like this. I think my best recommendation is to turn it off. I know if we should change it unless the upstream does. I think there are other rules that actually cover _use_ of subprocess.

Thanks for helping, actually I've turned it off, because I didn't find good reason to keep it

**Comment by charliermarsh** [comment]
Yeah I'm torn on these. They're _really_ strict, but they're also security-oriented rules, so it feels like it's against the intent to relax them. I'm gonna close for now, though I do empathize with the annoyance.
